### DISCUSSION TITLE
Why is AssertionError being deprecated in v8.0?

### DISCUSSION THREAD

**Discussion by adube** [openlayers/openlayers#14738] [Q&A] [answered]
Hello OpenLayers!  It's been a while :)

I'd be interested in learning the reasons behind the idea of deprecating / removing the AssertionError class in OpenLayers v8.0, for my personal knowledge.  I'm not against it, just want to learn the though process behind the idea.

Thanks,

Alexandre

**Comment by ahocevar** [comment, marked as answer]
See #14030. Currently, we have a mix of throwing errors directly and using `assert()`. To avoid confusion, the plan is to only go with the former in the future.
